cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/sidebar_filtering.spec.js

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+import {
+  DATASOURCE_NAME,
+  INDEX_PATTERN_NAME,
+  INDEX_NAME,
+  START_TIME,
+  END_TIME,
+  WORKSPACE_NAME,
+} from './constants.js';
+import * as dataExplorer from './helpers.js';
+import { SECONDARY_ENGINE, BASE_PATH } from '../../../../../utils/constants';
+
+const miscUtils = new MiscUtils(cy);
+
+const addFields = (testFields, expectedValues, pplQuery, sqlQuery, indexPattern = true) => {
+  const getDocTableHeaderByIndex = (index) => {
+    return cy.getElementByTestId('docTableHeaderField').eq(index);
+  };
+  const checkTableHeadersByArr = (expectedHeaders) => {
+    let currentHeader = 1;
+    expectedHeaders.forEach((header) => {
+      getDocTableHeaderByIndex(currentHeader).should('have.text', header);
+      currentHeader++;
+    });
+  };
+  const checkDocTableColumnByArr = (expectedValues, columnNumber) => {
+    let currentRow = 0;
+    expectedValues.forEach((value) => {
+      dataExplorer.getDocTableField(columnNumber, currentRow).should('have.text', value);
+      currentRow++;
+    });
+  };
+  const selectFields = (testFields) => {
+    testFields.forEach((field) => {
+      cy.getElementByTestId('fieldToggle-' + field).click();
+    });
+  };
+  if (indexPattern) {
+    dataExplorer.selectIndexPatternDataset(INDEX_PATTERN_NAME, 'DQL');
+    dataExplorer.setQueryEditorLanguage('DQL');
+    cy.setTopNavDate(START_TIME, END_TIME);
+  } else {
+    dataExplorer.selectIndexDataset(DATASOURCE_NAME, INDEX_NAME, 'OpenSearch SQL', 'timestamp');
+  }
```

Comment:
(no action needed). but just fyi, line 43-49 can be replaced by a custom command i am adding in my PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9112/files#diff-c71a49edcd56336affaeb1dfd1e47f29965ee58b7b9368943ea1468bb9fb307aR176 (minus the `setTopNavDate()`)